packages/legend-graph/src/models/protocols/pure/PureProtocolProcessorPlugin.ts

```diff
@@ -139,4 +140,11 @@ export abstract class PureProtocolProcessorPlugin extends AbstractPlugin {
    * This would provide a mechanism to add more elements in this reduced graph.
    */
   V1_getExtraExecutionInputGetters?(): V1_ExecutionInputGetter[];
+
+  /**
+   * Get the type of the inferred variable in property expression.
+   */
+  V1_getExtraPropertyExpressionTypeInferenceProcedures?(
```

Comment:
Lets follow the patterns of the other extensions. You should return a list of functions you can call that will return Type|undefined. Look at `V1_FunctionExpressionBuilder`